internal/extproc/translator/gemini_helper.go

```diff
@@ -623,7 +630,56 @@ func geminiLogprobsToOpenAILogprobs(logprobsResult genai.LogprobsResult) openai.
 	}
 }
 
-func buildGCPModelPathSuffix(publisher, model, gcpMethod string) string {
+// buildGCPModelPathSuffix constructs a path Suffix with an optional queryParams where each string is in the form of "%s=%s".
+func buildGCPModelPathSuffix(publisher, model, gcpMethod string, queryParams ...string) string {
 	pathSuffix := fmt.Sprintf("publishers/%s/models/%s:%s", publisher, model, gcpMethod)
+
+	// If there are no query parameters, return the path suffix as is.
+	queryParams = slices.Filter(queryParams[:0], queryParams, func(q string) bool { return q != "" })
```

Comment:
i don't think we want the code like this -- empty string is not a valid query param so it should be handled by a caller.